### PR TITLE
Updating to use the last version of hadoop

### PR DESCRIPTION
**Pull request by gustavolira**
This project is using too old version of Hadoop and is not working with JDG 7.3.CR1.
I just updated the code to use the current version of Hadoop. The tests still working

### CODE REVIEW

**Comment by karesti** [comment]
I think @gustavonalle might be the best to validate the correction

**Comment by karesti** [comment]
LGTM btw

File: hadoop/README.md [hunk 1]
```diff
@@ -46,16 +46,16 @@ Start the JDG Server
 Start Apache Flink
 --------------------
 
-1. Download and unpack v1.0 of [Apache Flink](https://flink.apache.org/downloads.html#binaries). On Windows, you can use 7-zip to extract the .tgz file.  
+1. Download and unpack v1.7 of [Apache Flink](https://flink.apache.org/downloads.html#binaries). On Windows, you can use 7-zip to extract the .tgz file.
```
Comment (gustavocoding) [inline hadoop/README.md]: Could you change this line to:
```
1. Download and unpack v1.7 of [Apache Flink with Hadoop](https://www-eu.apache.org/dist/flink/flink-1.7.1/flink-1.7.1-bin-hadoop28-scala_2.11.tgz). On Windows, you can use 7-zip to extract the .tgz file.  
```

Downloading the "Flink only server" will cause the job to fail

File: hadoop/pom.xml [hunk 2]
```diff
@@ -81,8 +81,21 @@
...
             <artifactId>flink-java</artifactId>
             <scope>provided</scope>
-            <version>0.9.1</version>
+            <version>1.7.1</version>
         </dependency>
+
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-core</artifactId>
+            <version>1.2.1</version>
```
Comment (gustavocoding) [inline hadoop/pom.xml]: This is a very old Hadoop version back from 2013. We should use Version 2.8

**Review by ghost — CHANGES_REQUESTED** [review]
I tried running the demo and got 
```
Caused by: java.lang.NoSuchMethodError: org.infinispan.client.hotrod.configuration.ConfigurationBuilder.balancingStrategy(Lorg/infinispan/client/hotrod/impl/transport/tcp/FailoverRequestBalancingStrategy;)Lorg/infinispan/client/hotrod/configuration/ConfigurationBuilder;
	at org.infinispan.hadoop.impl.InfinispanCache.getCache(InfinispanCache.java:59)
	at org.infinispan.hadoop.impl.InfinispanCache.getInputCache(InfinispanCache.java:47)
	at org.infinispan.hadoop.InfinispanInputFormat.createRecordReader(InfinispanInputFormat.java:41)
	at org.apache.flink.api.java.hadoop.mapreduce.HadoopInputFormatBase.open(HadoopInputFormatBase.java:186)
	at org.apache.flink.api.java.hadoop.mapreduce.HadoopInputFormatBase.open(HadoopInputFormatBase.java:59)
	at org.apache.flink.runtime.operators.DataSourceTask.invoke(DataSourceTask.java:170)
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:704)
	at java.lang.Thread.run(Thread.java:748)
```

Looks like the version of the Hadoop connector is to blame, since it's currently 
```
<version.jdg.hadoop>0.3.0.Final-redhat-9</version.jdg.hadoop>
```

Is should be ```0.4.0.Final-redhat-00001``` I think

File: hadoop/pom.xml (hunk 2 above)
Comment (gustavocoding) [reply hadoop/pom.xml]: Could you try with this version instead:
```xml
<dependency>
   <groupId>org.apache.hadoop</groupId>
   <artifactId>hadoop-mapreduce-client-core</artifactId>
   <version>3.1.1.redhat-00002</version>
</dependency>

Comment (gustavocoding) [reply hadoop/pom.xml]: I imported that a while ago into the MEAD repo

Comment (gustavolira) [reply hadoop/pom.xml]: thanks @gustavonalle , I just updated the code!

**Review by ghost — APPROVED** [review]

**Comment by gustavocoding** [comment]
Integrated, thanks @gustavolira !

**Pull request merged**